server/src/main/java/org/opensearch/cluster/service/applicationtemplates/TemplateRepositoryInfo.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service.applicationtemplates;
+
+/**
+ * The information to uniquely identify a template repository.
+ */
+public class TemplateRepositoryInfo {
+
+    private final String id;
+    private final long version;
+
```

Comment:
don't you need more details for identify as a repo like location etc. and also register a repo

Reply:
TemplateRepositoryMetadata